src/main/java/org/folio/circulation/domain/ItemAndRelatedRecords.java

```diff
@@ -0,0 +1,32 @@
+package org.folio.circulation.domain;
+
+public class ItemAndRelatedRecords {
```

Comment:
I'd prefer we didn't propagate this pattern further, it was naively introduced (by me) and has probably outlived it's usefulness.

Instead, please introduce a context specific structure, e.g. `InTransitReportEntry` or something similar, that can hold the state needed for an individual entry in the report. And build that up over time.

Reply:
@marcjohnson-kint What do you mean this pattern? Did I understand correctly that I need rename the class name from ItemAndRelatedRecords  to InTransitReportEntry?

Reply:
Apologies for being unclear. By this pattern I mean the `EntityAndRelatedRecords` naming pattern and widespread use. It was convenient at first, it has now become a bag for any associated state and makes the code harder to understand.

Yes, I’m practice I think this means renaming this class, to something that suits the usage better. `InTransitReportEntry` was only a suggestion, it depends what this set of objects is meant to represent?

Ideally, due to its specific use, it would also be separate (in a different package) to the rest of the domain. I realise that might be too much restructuring at this point.

Reply:
Renamed